src/Periphery/LiFiDEXAggregator.sol

```diff
@@ -732,6 +748,151 @@ contract LiFiDEXAggregator is WithdrawablePeriphery {
         uniswapV3SwapCallback(amount0Delta, amount1Delta, data);
     }
 
+    /// @notice Performs a swap through iZiSwap V3 pools
+    /// @dev This function handles both X to Y and Y to X swaps through iZiSwap V3 pools
+    /// @param stream [pool, direction, to]
+    /// @param from Where to take liquidity for swap
+    /// @param tokenIn Input token
+    /// @param amountIn Amount of tokenIn to take for swap
+    function swapIzumiV3(
+        uint256 stream,
+        address from,
+        address tokenIn,
+        uint256 amountIn
+    ) private {
+        address pool = stream.readAddress();
+        uint8 direction = stream.readUint8(); // 0 = Y2X, 1 = X2Y
+        address to = stream.readAddress();
+
+        if (from == msg.sender) {
+            IERC20(tokenIn).safeTransferFrom(
+                msg.sender,
+                address(this),
+                amountIn
+            );
+        }
+
+        lastCalledPool = pool;
+
+        if (direction == DIRECTION_TOKEN0_TO_TOKEN1) {
+            IiZiSwapPool(pool).swapX2Y(
+                to,
+                uint128(amountIn),
+                IZUMI_LEFT_MOST_PT,
+                abi.encode(tokenIn)
+            );
+        } else {
+            IiZiSwapPool(pool).swapY2X(
+                to,
+                uint128(amountIn),
+                IZUMI_RIGHT_MOST_PT,
+                abi.encode(tokenIn)
+            );
+        }
+
+        if (lastCalledPool != IMPOSSIBLE_POOL_ADDRESS) {
+            revert IzumiV3SwapUnexpected();
+        }
```

Comment:
Feel like this could check could already be done before we attempt the swap, right?

Reply:
Think it cant...the callback function for this swap can potentially alter this flag. I left it where it was